disk-buffering/src/test/java/io/opentelemetry/contrib/disk/buffering/internal/storage/files/WritableFileTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.internal.storage.files;
+
+import static io.opentelemetry.contrib.disk.buffering.internal.storage.TestData.MAX_FILE_AGE_FOR_WRITE_MILLIS;
+import static io.opentelemetry.contrib.disk.buffering.internal.storage.TestData.MAX_FILE_SIZE;
+import static io.opentelemetry.contrib.disk.buffering.internal.storage.files.utils.Constants.NEW_LINE_BYTES_SIZE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import io.opentelemetry.contrib.disk.buffering.internal.storage.TestData;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.exceptions.NoSpaceAvailableException;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.exceptions.ResourceClosedException;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.exceptions.WritingTimeoutException;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.utils.TimeProvider;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class WritableFileTest {
```

Comment:
not sure if I missed it, but I couldn't find any integration test that tries to test all components together.
such a test could also reduce the number of required unit tests without hurting coverage.

Reply:
You're right, for now I only focused on unit tests. Though I think we can add an integration one before handing it over for review later. If you have some ideas of best practices for these kinds of tests I'd be glad to get some insights from your side since I'm currently too into Android and Gradle plugins integration test suites, so I'm not aware of the latest tools for other environments that might make these easier to create.

Reply:
such an "integration test" can be called like a unit test - no special gradle plugin or similar is needed.

The integration test should cover most classes that make the core logic are tested (and not mocked) - in our case that would be the entire storage layer. 
The integration test would first create files (using the exporters) and then read (and possibly delete) the same files using the export loop.
Technically, you can create the required object graph with constructor injection - like you're already doing in some tests.

Reply:
Sounds simple enough.

Reply:
Thank you, I've made the changes.